WordPress/Classes/Networking/LoginClient.swift

```diff
@@ -78,4 +81,134 @@ actor LoginClient {
+
+    var onLoginComplete: (() -> Void)?
+
+    init(loginClient: LoginClient) {
+        self.loginClient = loginClient
+        super.init()
+    }
+
+    override func startLogin() {
+        debugPrint("Attempting login with \(super.urlField)")
+        Task {
+            do {
+                let authUrl = try await self.buildLoginUrl(from: super.urlField)
+                debugPrint("Presenting Login Page \(authUrl)")
+
+                let urlWithToken = try await withUnsafeThrowingContinuation { continuation in
+                    let session = ASWebAuthenticationSession(url: authUrl, callbackURLScheme: "x-wpcom-login") { url, error in
+
+                        if let url {
+                            continuation.resume(returning: url)
+                        }
+
+                        if let error {
+                            continuation.resume(throwing: error)
+                        }
+                    }
+                    session.presentationContextProvider = self.presentationContextProvider
+                    session.start()
+                }
+
+                let credentials = try SelfHostedLoginDetails.fromApplicationPasswordResponse(urlWithToken)
+                let blogOptions = try await loadSiteOptions(details: credentials)
+
+                // Only store the new site after credentials are validated.
+                let _ = try await Blog.createRestApiBlog(with: credentials, in: ContextManager.shared)
+
+                // Automatically show the new site.
+                assert(WordPressAuthenticator.shared.delegate != nil)
+                let wporg = WordPressOrgCredentials(
+                    username: credentials.username,
+                    password: credentials.password,
+                    xmlrpc: credentials.derivedXMLRPCRoot.absoluteString,
+                    options: blogOptions
+                )
+                WordPressAuthenticator.shared.delegate!.sync(credentials: .init(wporg: wporg)) {
+                    NotificationCenter.default.post(name: Foundation.Notification.Name(rawValue: WordPressAuthenticator.WPSigninDidFinishNotification), object: nil)
+                }
+
+                self.onLoginComplete?()
+            } catch let err {
+                debugPrint(err.localizedDescription)
+            }
+        }
+    }
+
+    private func buildLoginUrl(from userInput: String) async throws -> URL {
+        let result = try await self.loginClient.performLoginAutodiscovery(for: userInput)
+        var mutableAuthURL = URL(string: result.loginUrl.url())!
+
+        var appNameValue = AppConfiguration.isJetpack ? "Jetpack iOS" : "WordPress iOS"
+
+        #if os(macOS)
+        if let deviceName = Host.current().localizedName {
+            appNameValue += " - \(deviceName)"
+        }
+        #else
+        let deviceName = await UIDevice.current.name
+        appNameValue += " - \(deviceName)"
+        #endif
+
+        if #available(iOS 16.0, *) {
```

Comment:
FYI I'm bumping the deployment target here https://github.com/wordpress-mobile/WordPress-iOS/pull/23476. But I couldn't get the CI to build it.